src/modules/Elsa.MongoDB/Features/MongoDbFeature.cs

```diff
@@ -0,0 +1,117 @@
+using WorkflowExecutionLogRecord = Elsa.Workflows.Runtime.Entities.WorkflowExecutionLogRecord;
+
+namespace Elsa.MongoDB.Features;
+
+/// <summary>
+/// Configures MongoDb.
+/// </summary>
+public class MongoDbFeature : FeatureBase
+{
+    /// <inheritdoc />
+    public MongoDbFeature(IModule module) : base(module)
+    {
+    }
+
+    /// <summary>
+    /// A delegate that configures MongoDb.
+    /// </summary>
+    public Action<MongoDbOptions> Options { get; set; } = _ => { };
+
+    /// <inheritdoc />
+    public override void Apply()
+    {
+        Services.Configure(Options);
+
+        Services.AddSingleton(CreateDatabase);
+        
+        RegisterSerializers();
+
+        AddMongoCollection<Application>(Services, "applications");
+        AddMongoCollection<User>(Services, "users");
+        AddMongoCollection<Role>(Services, "roles");
+        AddMongoCollection<Label>(Services, "labels");
+        AddMongoCollection<WorkflowDefinitionLabel>(Services, "workflow_definition_labels");
+        AddMongoCollection<WorkflowDefinition>(Services, "workflow_definitions");
+        AddMongoCollection<WorkflowInstance>(Services, "workflow_instances");
+        AddMongoCollection<WorkflowState>(Services, "workflow_states");
+        AddMongoCollection<WorkflowExecutionLogRecord>(Services, "workflow_execution_logs");
+        AddMongoCollection<StoredTrigger>(Services, "triggers");
+        AddMongoCollection<StoredBookmark>(Services, "bookmarks");
+
+        Services
+            .AddSingleton<IApplicationStore, MongoApplicationStore>()
+            .AddSingleton<IUserStore, MongoUserStore>()
+            .AddSingleton<IRoleStore, MongoRoleStore>()
+            .AddSingleton<ILabelStore, MongoLabelStore>()
+            .AddSingleton<IWorkflowDefinitionLabelStore, MongoWorkflowDefinitionLabelStore>()
+            .AddSingleton<IWorkflowDefinitionStore, MongoWorkflowDefinitionStore>()
+            .AddSingleton<IWorkflowInstanceStore, MongoWorkflowInstanceStore>()
+            .AddSingleton<IWorkflowExecutionLogStore, MongoWorkflowExecutionLogStore>()
+            .AddSingleton<IWorkflowStateStore, MongoWorkflowStateStore>()
+            .AddSingleton<ITriggerStore, MongoTriggerStore>()
+            .AddSingleton<IBookmarkStore, MongoBookmarkStore>()
+            .AddHostedService<CreateIndices>()
+            .AddHealthChecks();
+    }
+
+    private void RegisterSerializers()
+    {
+        BsonSerializer.RegisterSerializer(typeof(object), new PolymorphicSerializer());
+    }
+
+    private static IMongoDatabase CreateDatabase(IServiceProvider sp)
+    {
+        var options = sp.GetRequiredService<IOptions<MongoDbOptions>>().Value;
+        var settings = MongoClientSettings.FromConnectionString(options.ConnectionString);
+
+        settings.ClusterConfigurator = cb => cb.Subscribe(new DiagnosticsActivityEventSubscriber());
+        settings.ApplicationName = "elsa_workflows";
```

Comment:
Would it be better to only provide this as a default value in case ApplicationName was not specified from the connection string? Right now, it would overwrite it always.